### ISSUE TITLE
404 error in Laravel 9

### ISSUE THREAD

**Issue by bhartvigsen** [motze92/office365-mail#29]
Installed, created relevant tokens and secret, tried to send test email, I get:

GuzzleHttp\Exception\ClientException with message 'Client error: `POST https://login.microsoftonline.com//oauth2/v2.0/token` resulted in a `404 Not Found` response'

**Comment by motze92** [comment]
It seems you didn't set OFFICE365MAIL_TENANT correctly in your .env

**Comment by bhartvigsen** [comment]
I do not believe that to be the case.

In my .env I have configured:
MAIL_MAILER=office365mail

and
OFFICE365MAIL_CLIENT_ID
OFFICE365MAIL_TENANT
OFFICE365MAIL_CLIENT_SECRET

Using values in the App Registrations section. In the "Essentials" block for the CLIENT_ID and TENANT, and I generated a secret in the 'Certificates & Secrets' page and copied the value from there.

Possible that there is something else I need to do? I followed the instructions from packagist pretty much verbatim.

EDIT: Digging into the code it looks like the call to config() is failing to return the values in .env. This is a relatively new Laravel 9 install. Not sure where the issue lies, and may not be an issue with the lib at all, but I'm curious why this would fail on a fresh app install. I've tried the usual artisan config:clear / cache:clear / dump-autoload etc.

EDIT2: Swapping your calls to config() with env() and replacing the dot notation with the key values from .env appears to work.... getting a 401 unauthorized error now.... continuing to investigate...I see it is retrieving an access token successfully in the getAccessToken() call, so the 401 must be from something else going on

EDIT3: It appears the 401 response from MS is occurring in the $client->request() call at line 316 in GraphRequest.php. When I inspect the $client object I see a guzzle client that appears to be fully configured, and I see the access token in there, so I'm a little stumped as to what is failing. Error:

"error":{"code":"NoPermissionsInAccessToken","message":"The token contains no permissions, or permissions can not be understood."

EDIT4: Probably final update, I had previously added the Mail.Send permission but when I went back and reviewed it I see a status that shows 'Not granted for ..' so I believe this is an organizational permissions issue rather than an issue with the lib. So in summary, I probably 'fixed' it but I can't validate that at this point. I have a hunch your call to config() may need to be revised for some change that may have occurred in Laravel 9 though.

I notice your instructions for Laravel <7 and >7 are the same on packagist btw
